### PR TITLE
Implement CancellationToken in Task ServerEventsController.SSE.

### LINKED ISSUE #5: Server doesn't respond to ^C
**Description**
The server doesn't respond immideately to a ^C from the command line.

**Reproduction**
1. Open a console and cd into the repo
2. Type `dotnet run` and return
3. Select the correct serial port
4. Open a browser window at [`https://localhost:7097/api/sse`](https://localhost:7097/api/sse) to actually start-up the server
5. Go back to the console and press Ctrl+C (or Cmd+C on a Mac)
6. The server doesn't shutdown, or only does it after a long time

**Expected behavior**
When sendind ^C, the server should quickly react and properly shutdown

**System info**
 - OS: Windows
 - .NET version: `8.0.400`

**Possible causes**
Most likely, this bug is caused by the infinite loop in [`ServerEventsController.SSE`](https://github.com/orbisat-oeiras/backend/blob/4d3a6a5bcb7279a2399e7d5bc6c81a83e758de06/Controllers/ServerEventsController.cs#L65). Presumably, this loop should handle some kind of `CancellationToken` (see [this article](https://hackernoon.com/why-do-you-need-a-cancellation-token-in-c-for-tasks) for more info). See [this answer](https://stackoverflow.com/a/58565850/22222542) for a possible solution.

**Additional context**
On a broader level, the current communication scheme between the backend and the frontend should be revised, as well as the SSE implementation. However, that's a discussion for somewhere else.

### PR DESCRIPTION
**Pull request by ordinho**
Will close #5

### CODE REVIEW

File: Controllers/ServerEventsController.cs [hunk 1]
```diff
@@ -61,8 +60,9 @@ public async Task SSE() {
             }
 
 			// Keep the server alive
-			// This feels very dodgy
-            while (true) {
+			// This still feels very dodgy
+			
+            while (cancellationToken == default) {
```
Comment (levimcgomes) [inline Controllers/ServerEventsController.cs:65]: Hmm, comparing `cancellationToken` to `default` seems dodgy... Any special reasons? Why not use `cancellationToken.IsCancellationRequested`? It does work, though!

File: Controllers/ServerEventsController.cs [hunk 2]
```diff
@@ -28,13 +28,12 @@ public ServerEventsController(ILogger<ServerEventsController> logger, IEnumerabl
...
 			_logger.LogInformation("DI provided {evtFinalizerCount} event finalizers.", _eventFinalizers.Count().ToString());
 		}
-
 		/// <summary>
 		/// Provide a GET endpoint for connecting to the SSE channel
 		/// </summary>
 		/// <returns>Good question</returns>
 		[HttpGet()]
-		public async Task SSE() {
+		public async Task SSE(CancellationToken cancellationToken = default) {
```
Comment (levimcgomes) [inline Controllers/ServerEventsController.cs:36]: Is it necessary to make this an optional argument? In fact, should it be an optional agrument?

**Review by levimcgomes — CHANGES_REQUESTED** [review]

File: Controllers/ServerEventsController.cs (hunk 1 above)
Comment (ordinho) [reply Controllers/ServerEventsController.cs:65]: I also tried using `while (!cancellationToken.IsCancellationRequested)` but it didn't work at all. I really just went by whether it worked or not.

File: Controllers/ServerEventsController.cs (hunk 2 above)
Comment (ordinho) [reply Controllers/ServerEventsController.cs:36]: You're right, omitting the `= default` doesn't seem to change anything, thanks.

File: Controllers/ServerEventsController.cs (hunk 1 above)
Comment (levimcgomes) [reply Controllers/ServerEventsController.cs]: Turns out that infinite loop wasn't necessary at all! I've removed it, as well as the `CancellationToken` shenanigans, and it seems to be working fine.

**Review by levimcgomes — APPROVED** [review]

**Comment by levimcgomes** [comment]
Judging from my commit's diff, our editors don't agree on formatting. I was going to add some auto-formatting and the likes anyway, so it won't be much of an issue moving forward.

**Comment by ordinho** [comment]
Noticed that too, great idea.

**Pull request merged**